app/code/GumennoyEL/FirstModule/etc/frontend/routes.xml

```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:App/etc/routes.xsd">
+    <router id="standard">
+        <route id="home_work" frontName="home-work">
```

Comment:
Having different names for route configuration id and front name could cause confusion. Probably return $this->getUrl('home_work/jsonresponse/index'); in your block app/code/GumennoyEL/FirstModule/Block/CustomBlock.php is incorrect (I'm not sure, please confirm if it's working)
My advice is to use the same id without _ characters. Use homework